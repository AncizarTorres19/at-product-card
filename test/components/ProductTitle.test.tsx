import React from 'react'
import renderer from 'react-test-renderer'
import { ProductTitle, ProducCard } from '../../src/components';
import { product1 } from '../data/products';

describe('ProductTitle', () => {
    test('debe de mostrar el componente correctamente con el titulo perzonalizado', () => {
        const wrapper = renderer.create(<ProductTitle title="Titulo personalizado" className='custom-class' />);
        // console.log(wrapper.toJSON());
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('debemos mostrar el componente con el nombre del producto', () => {
        const wrapper = renderer.create(
            <ProducCard
                product={product1}
            >
                {
                    () => (
                        <ProductTitle />
                    )
                }
            </ProducCard>
        );
        // console.log(wrapper.toJSON());
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('debemos mostrar el componente con el nombre del producto', () => {
        const wrapper = renderer.create(
            <ProducCard
                product={product1}
            >
                {
                    () => (
                        <>
                            <ProductTitle />
                        </>
                    )
                }
            </ProducCard>
        );
        // console.log(wrapper.toJSON());
        expect(wrapper.toJSON()).toMatchSnapshot();
    })
});