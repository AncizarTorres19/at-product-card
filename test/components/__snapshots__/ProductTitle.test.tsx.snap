// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle debe de mostrar el componente correctamente con el titulo perzonalizado 1`] = `
<span
  className="productDescription custom-class"
>
  Titulo personalizado
</span>
`;

exports[`ProductTitle debemos mostrar el componente con el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Without Image
  </span>
</div>
`;

exports[`ProductTitle debemos mostrar el componente con el nombre del producto 2`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Without Image
  </span>
</div>
`;
